scripts/hic/qc_hic.py

```diff
@@ -0,0 +1,153 @@
+import os
+import argparse
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description="QC a Hi-C experiment.")
+    parser.add_argument(
+        "--all_valid_pairs_stats", type=str, help="All valid pairs stats file."
+    )
+    parser.add_argument("--pairing_stats", type=str, help="Pairing stats file.")
+    parser.add_argument(
+        "--mapping_stats_read1", type=str, help="Mapping stats for read 1."
+    )
+    parser.add_argument(
+        "--mapping_stats_read2", type=str, help="Mapping stats for read 2."
+    )
+    parser.add_argument("--fithichip_bed", type=str, help="FithiChIP bed file.")
+    parser.add_argument("--fithichip_q01_bed", type=str, help="FithiChIP q01 bed file.")
+    parser.add_argument("--peaks_bed", type=str, help="Peaks bed file.")
+    parser.add_argument("--prefix", type=str, help="Prefix for output file.")
+
+    args = parser.parse_args()
+
+    return args
+
+
+if __name__ == "__main__":
+    args = get_args()
+    FILES = [
+        args.all_valid_pairs_stats,
+        args.pairing_stats,
+        args.mapping_stats_read1,
+        args.mapping_stats_read2,
+    ]
+    VARIABLE = [
+        "valid_interaction",
+        "cis_shortRange",
+        "cis_longRange",
+        "Total_pairs_processed",
+        "Reported_pairs",
+        "total_R2",
+        "mapped_R2",
+        "total_R1",
+        "mapped_R1",
+    ]
+    RESULTS = {}
+    for eachfile in FILES:
+        with open(eachfile) as f:
+            for line in f:
+                array = line.split()
+                if array[0] in VARIABLE:
+                    RESULTS[array[0]] = int(array[1])
+
+    PERCENTAGES = {}
+    COUNTS = {}
+    VERDICT = {}
+    REP_VAR = [
+        "R1_aligned",
+        "R2_aligned",
+        "valid_interactionPairs",
+        "cis_shortRange",
+        "cis_longRange",
+    ]
+
+    if args.fithichip_bed is not None and os.path.isfile(args.fithichip_bed):
+        with open(args.fithichip_q01_bed) as fithichip:
+            LOOPS_SIGNIFICANT = len(fithichip.readlines()) - 1
+        with open(args.fithichip_bed) as fithichip:
+            LOOPS = len(fithichip.readlines()) - 1
+    if args.peaks_bed is not None and os.path.isfile(args.peaks_bed):
+        with open(args.peaks_bed) as peaks:
+            PEAKS = len(peaks.readlines()) - 1
```

Comment:
1. bug if `fithichip_bed` is defined but `_q01_bed` isn't
2. Are any of these files large? They are being completely loaded into memory just to do a `wc -l` check. If they're always small I guess that's fine...
3. Move these to the outreport writing blocks. They aren't used during any calculation and that will keep us from having to silence `pyright`

Reply:
Also, does the `-1` seem correct to you? `len(readlines())` should contain the number of lines. The only reason to subtract 1 would be if the BED file contained a header line.

Reply:
🤷 pull up an example and see if it has a header

Reply:
I didn't implement any of the other HiChIP steps. So I don't have an example file...I debated leaving this out of this script, but thought that would be more problematic if we ever implemented HiChIP processing.